source/aggregation-pipeline-builder.txt

```diff
@@ -89,69 +89,226 @@ Limitations
+
+              - Windows / Linux: :guilabel:`Ctrl + Shift + A`
+
+              - Mac: :guilabel:`⌘ + Shift + A`
+            
+            - To add a stage before the current stage: 
+              
+              - Windows / Linux: :guilabel:`Ctrl + Shift + B`
+
+              - Mac: :guilabel:`⌘ + Shift + B` 
+
+         .. step:: Navigate between stages. 
+            
+            To navigate between different stages, select the stage you 
+            want to edit from the :guilabel:`Stage` dropdown in the upper-right 
+            corner of the Focus Mode modal. 
+
+            You can navigate between stages with the following keyboard 
+            shortcuts: 
+
+            - To go to the stage before the current stage: 
+
+              - Windows / Linux: :guilabel:`Ctrl + Shift + 9`
+
+              - Mac: :guilabel:`⌘ + Shift + 9`
+            
+            - To add a stage before the current stage: 
+              
+              - Windows / Linux: :guilabel:`Ctrl + Shift + 0`
+
+              - Mac: :guilabel:`⌘ + Shift + 0` 
+
+         .. step:: Run the pipeline.
+            
+            Click :guilabel:`x` to exit Focus Mode and select 
+            :guilabel:`Run` at the top right of the pipeline builder. 
+            |compass-short| returns your results in the document view. 
+
+   .. tab:: Text View Mode 
+      :tabid: text 
+
+      .. procedure::
+         :style: normal
+         
+         .. step:: Select the :guilabel:`Text` view.
+            
+            In the aggregation pipeline pane, click the 
+            :guilabel:`</> Text` toggle switch to enable text mode for pipeline 
+            editing.
+            
+            .. figure:: /images/compass/agg-builder-textview-toggle-on.png
+               :alt: Toggle textmode on
+         
+         .. step:: Enter your aggregation pipeline.
+            
+            Enter valid aggregation syntax in the text editor. The 
+            following example pipeline limits the query results to ``4`` 
+            documents.
+            
+            .. code-block:: javascript
+               
+               [ { "$limit" : 4 } ]
+               
+            .. note:: 
+               
+               **(Optional)** If preview mode is enabled, you can click 
+               :guilabel:`Output Options` in the Pipeline Output pane to expand 
+               all nested fields and documents within the preview results.
+               
+               .. figure:: /images/compass/agg-builder-textview-expand-pipeline-preview.png
+                  :alt: Pipeline Output preview
+               
+         .. step:: Run the pipeline.
+            
+            Click :guilabel:`Run` at the top right of the pipeline 
+            builder. |compass-short| returns your results in the document view. 
+            
+            .. note:: 
+               
+               **(Optional)** Click the :guilabel:`Output Options` dropdown 
```

Comment:
Same idea here as my other comment about ordering this like "To (x), (do y).":

To expand all embedded fields and documents within the query results, click the :guilabel:`Output Options` dropdown and select :guilabel:`Expand all fields`.